Python/optimizer.c

```diff
@@ -1280,7 +1280,14 @@ uop_optimize(
 {
     _PyBloomFilter dependencies;
     _Py_BloomFilter_Init(&dependencies);
-    _PyUOpInstruction buffer[UOP_MAX_TRACE_LENGTH];
+    _PyThreadStateImpl *tstate = (_PyThreadStateImpl *)_PyThreadState_GET();
+    if (tstate->jit_uop_buffer == NULL) {
+        tstate->jit_uop_buffer = (_PyUOpInstruction *)PyMem_RawMalloc(UOP_MAX_TRACE_LENGTH*sizeof(_PyUOpInstruction));
```

Comment:
This is not thread-safe on FT. But we can fix it later I guess.

Reply:
> It is, right? Because it's on the thread state. But I want to move it to the interpreter state, so it would need either a lock (or to not be lazily created).

Oh wait I mixed up, yes this is safe. Thread state is safe, not interpreter state. My bad.

Reply:
IIUC, but even for the interpreter state, the buffer will only be accessed by the single optimizer per interpreter?

Reply:
Since the per-thread model is already thread-safe in the current structure, I’d prefer to keep this buffer in the thread state. Otherwise, if it’s stored in the interpreter state, we’d need to lock the buffer, which makes the codebase less seamless. But please tackle me if I know something wrong.